app/src/main/java/me/hufman/androidautoidrive/carapp/carinfo/CarDetailedInfo.kt

```diff
@@ -135,41 +138,52 @@ class CarDetailedInfo(carCapabilities: Map<String, Any?>, cdsMetrics: CDSMetrics
 
 	// windows are not translated
 	fun formatWindowState(name: String, state: CDSMetrics.WindowState): String = when(state.state) {
-		CDSMetrics.WindowState.State.CLOSED -> "$name Closed"
-		CDSMetrics.WindowState.State.TILTED -> "$name Tilted"
-		CDSMetrics.WindowState.State.OPENED -> "$name Opened ${state.position}%"
+		CDSMetrics.WindowState.State.CLOSED -> "$name ${L.CARINFO_WINDOW_CLOSED}"
+		CDSMetrics.WindowState.State.TILTED -> "$name ${L.CARINFO_WINDOW_TILTED}"
+		CDSMetrics.WindowState.State.OPENED -> "$name ${L.CARINFO_WINDOW_OPENED} ${state.position}%"
 	}.trimStart()
 	fun Flow<CDSMetrics.WindowState>.format(name: String): Flow<String> = this.map {
 		formatWindowState(name, it)
 	}
 
-	val sunroof = cdsMetrics.sunroof.format("Sunroof")
+	val sunroof = cdsMetrics.sunroof.format("${L.CARINFO_SUNROOF}")
 	val windowDriverFront = cdsMetrics.windowDriverFront.format("")
 	val windowPassengerFront = cdsMetrics.windowPassengerFront.format("")
 	val windowDriverRear = cdsMetrics.windowDriverRear.format("")
 	val windowPassengerRear = cdsMetrics.windowPassengerRear.format("")
 
 	// gps fields
 	val countryLabel = cdsMetrics.gpsCountry
-	val cityLabel = cdsMetrics.gpsCity.map { "City: $it" }
+	val cityLabel = cdsMetrics.gpsCity //.map { "City: $it" }
 	val streetLabel = cdsMetrics.gpsStreet.combine(cdsMetrics.gpsHouseNumber) { street, houseNumber ->
 		"$houseNumber $street".trim()
 	}
 	val crossStreetLabel = cdsMetrics.gpsCrossStreet.map {
 		if (it.isBlank()) "" else "and $it"
 	}
-	val altitudeLabel = cdsMetrics.gpsAltitude.map { "Altitude: ${it}m" }
-	val latitudeLabel = cdsMetrics.gpsLat.map { "Lat: $it" }
-	val longitudeLabel = cdsMetrics.gpsLon.map { "Long: $it" }
-	val rawHeading = cdsMetrics.rawHeading.format("%.0f Raw Heading")
+	val altitudeLabel = cdsMetrics.gpsAltitude.map { "${it}m ${L.CARINFO_GPSALTITUDE}" }
+	val latitudeLabel = cdsMetrics.gpsLat.map { "$it ${L.CARINFO_GPSLATITUDE}" }
+	val longitudeLabel = cdsMetrics.gpsLon.map { "$it ${L.CARINFO_GPSLONGITUDE}" }
+//	val rawHeading = cdsMetrics.rawHeading.format("%.0f Raw Heading")
+//	val ACActualPower =cdsMetrics.ACCompressorActualPower.format("%.0f").map { "$it Power" }
+//	val ACDualmode =cdsMetrics.ACCompressorDualMode.format("%.0f").map { "$it Dualmode" }
+//	val ACActualTorque =cdsMetrics.ACCompressorActualTorque.format("%.0f").map { "$it Torque" }
```

Comment:
I think these AC power metrics might be useful in the HVAC screen, perhaps in an Advanced version of the HVAC screen. Feel free to uncomment these and add them to the HVAC screen.

Reply:
I can do this...but I can*t test because value for ACPower and ACDualmode are constant in my car (ACPower 255, Dualmode 3 (=invalid). I think this is an option for pure electric cars like BMW i-Series. In electric cars, climate compressors can also be used for heating, this is more efficient heating.
I'll add those items to my climate data screen but only showing if there are sufficient values from car (AC Power <255, Dualmode 1 (on) or 2(off). See screenshot from decompiled code in myBMW App 3.3:
![grafik](https://github.com/BimmerGestalt/AAIdrive/assets/81849700/25e18cd3-00ad-42fd-a034-7d0ce0742365)

Reply:
Done. Note ACActualTorque is mapped to ACCompressorLevel in CDSMetrics, so it is not neccessary here.

Reply:
That makes sense, most of these metrics have the `takeIf` thing to only emit if they are not invalid. I'll try it out soon and tell you what my electric car says :)